internal/plans/planfile/tfplan_test.go

```diff
@@ -325,10 +325,12 @@ func TestTFPlanRoundTrip(t *testing.T) {
 		t.Fatal(err)
 	}
 	{
-		// remove ephemeral from the plan since that should have been skipped from been written
-		plan.Changes.Resources = slices.DeleteFunc(plan.Changes.Resources, func(src *plans.ResourceInstanceChangeSrc) bool {
+		// nullify the ephemeral values from the initial plan since those must be nil in the plan file
+		i := slices.IndexFunc(plan.Changes.Resources, func(src *plans.ResourceInstanceChangeSrc) bool {
```

Comment:
Why are we doing this in a _test file?  To replicate other functionality?

Reply:
Exactly! Because we are sending a change with values to the function that writes the plan, and when we read it, we want to double check that the information didn't end up in the file.